tests/soter/test_vectors/soter_kdf.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 Cossack Labs Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"fmt"
+	"strings"
+)
+
+var key = []byte{0xAD, 0xEA, 0xDB, 0xEE, 0x01, 0x02, 0x03, 0x04}
+var label = "ZRTP KDF test vectors"
+var context = []byte{0x43, 0x83, 0xF3, 0x27, 0x0E, 0x11}
+var lengths = []int{1, 2, 4, 8, 16, 32}
+
+// A la RFC 6189 with HMAC-SHA-256
+// KDF(KI, label, context, L) = HMAC(KI, i || label || 0x00 || context || L)
+func zrtpKDF(key []byte, label string, context []byte, length int) []byte {
+	if !(1 <= length && length <= 32) {
+		panic("invalid length")
+	}
+	data := make([]byte, 0, len(label)+len(context)+5)
+	data = append(data, []byte{0x00, 0x00, 0x00, 0x01}...)
+	data = append(data, []byte(label)...)
+	data = append(data, 0x00)
+	data = append(data, context...)
+	// Here ZRTP KDF appends "length", but Soter KDF omits it
+	mac := hmac.New(sha256.New, key)
+	mac.Write(data)
+	res := mac.Sum(nil)
+	return res[:length]
+}
+
+func hexString(data []byte) string {
```

Comment:
why not just to use https://golang.org/pkg/encoding/hex/#EncodeToString ?

Reply:
Cool, I did not know about that. Go utils are amazing at times. I'll update the code to use it.

Reply:
Oh... on the second thought. We need a custom function to output C-style escaped hex bytes.

```go
hex.EncodeToString([]byte{1, 2, 3})
```

outputs `010203` while we need to write `\x01\x02\x02` for C code to parse that correctly.

I did not find a suitable function for that quickly, so I guess a simple utility is okay.

Reply:
thanks for explanation